pyspeckit/spectrum/plotters.py

```diff
@@ -378,14 +402,13 @@ def label(self, title=None, xlabel=None, ylabel=None, verbose_label=False,
 
         if xlabel is not None:
             self.xlabel = xlabel
-        elif isinstance(self.Spectrum.xarr.xtype,str):
-            self.xlabel = self.Spectrum.xarr.xtype.title()
+        elif self._xunit:
+            self.xlabel += " ("+u.Unit(self._xunit).to_string()+")"
             if verbose_label:
                 self.xlabel = "%s %s %s" % ( self.Spectrum.xarr.velocity_convention.title(),
-                        self.Spectrum.xarr.frame.title(),
+                        # self.Spectrum.xarr.frame.title(),
```

Comment:
removed, wow! there are so many factors to take into account! Also, it is the first time I realise the solar system has a center of mass :open_mouth: